wordpress-mobile/WordPress-Android
# Enforce western arabic numerals for Stats

Fixes Partially #17002 — Handles only the Stats screens and the Today Stats card on My Site Dashboard

This PR introduces a wrapper over `MaterialTextView` that enforces the use of Western Arabic numerals by replacing Eastern Arabic and Persian/Urdu numerals to their Western Arabic counterparts.

This UI component was then applied to every stats widget which uses `MaterialTextView` and displays numerals.

The purpose of this PR is to make sure we're always displaying Western Arabic numerals on the Stats screens, to mainly address the remark from the issue:
> _This looks especially strange on screens where we use both types of numerals_

To test:

##### 1️⃣ Test Case 1 : WordPress App Stats

1. Open the WordPress App, Go to `Me` → `App Settings` → `Interface Language` and select `Arabic`.
2. Go back to `My Site` → `Home` tab
   - Expect the numbers on `Today's Stats` card to be western arabic.
3. Go to `Stats` & explore the first tab (Insights)
   - Expect all displayed numbers to be western arabic.
4. Go to any other `Stats` tab (Days/Weeks/...)
   - Expect all displayed numbers to be western arabic.

##### 1️⃣ Test Case 2 : Jetpack App Stats

1. Open the Jetpack App, Go to `Me` → `App Settings` → `Interface Language` and select `Arabic`.
2. Repeat steps 2-4 from Test Case 1
3. Tap "View More" on one of the stats cards having a different value than `0`
   - Expect numbers to be Western Arabic
   - Expect the numeral inside the Pie Chart to be of larger font size than the word `Totals`

<details>
<summary>Preview Test Case 2 — Step 3</summary>

<img width="314" src="https://user-images.githubusercontent.com/4588074/190453085-8e028195-a23e-450f-a379-770ba2997ede.png">

**Note** For easier understanding the screenshot is in English. Same result should be expected in Arabic.

---

</details>

### Previews

#### Stats → Insights tab
| Before | After |
| --- | --- |
| ![stats_insights](https://user-images.githubusercontent.com/4588074/190380852-ef802513-ecaa-4b67-8304-9566630dc24b.png) | ![stats_followers_fix](https://user-images.githubusercontent.com/4588074/190380863-99f51981-ce97-4d7b-a7ca-dfa7c0dc5fe4.png) |

## Regression Notes
1. Potential unintended areas of impact
   Unexpected, since we're only replacing Indo-Arabic and Persian/Urdu numerals with the normal (Western) Arabic numerals.

6. What I did to test those areas of impact (or what existing automated tests I relied on)
   Tested manually in Chinese & English languages.

7. What automated tests I added (or what prevented me from doing so)
   N/a, we're only touching the view layer with these changes.

PR submission checklist:
